src/System.Windows.Forms/src/System/Windows/Forms/TreeNodeCollection.cs

```diff
@@ -367,6 +367,18 @@ private int AddInternal(TreeNode node, int delta)
 
             // If the TreeView is sorted, index is ignored
             TreeView tv = owner.TreeView;
+
+            if (tv is not null)
+            {
+                KeyboardToolTipStateMachine.Instance.Hook(node, tv.KeyboardToolTip);
+                List<TreeNode> nodes = new List<TreeNode>();
+                node.GetChildNodes(nodes);
+                foreach (TreeNode treeNode in nodes)
```

Comment:
In addition to the comment above:

```suggestion
                foreach (TreeNode treeNode in node.GetChildNodes())
```